README.md

```diff
@@ -35,4 +42,33 @@ craftguide.register_craft({
 })
 ```
 
-![Preview2](https://i.imgur.com/bToFH38.png)
+#### Progressive mode
+
+##### `craftguide.progressive_filter_recipes(recipes, player)`
+
+This function is used to filter recipes when progressive mode is enabled. It can
+be overridden to change the recipes that are normally displayed.
+
+The function should return the recipes to be displayed, given the available
+recipes and an `ObjectRef` to the craft guide user. Each recipe is a table of
+the form returned by `minetest.get_craft_recipe`.
+
+Example function to hide recipes from a mod called "secretstuff":
+
+```lua
+function craftguide.progressive_filter_recipes(recipes, player)
+    local filtered = {}
+    for _, recipe in ipairs(recipes) do
+        if not (recipe.output:sub(1, 12) == "secretstuff:") then
+            filtered[#filtered + 1] = recipe
+        end
+    end
+    return filtered
+end
+```
+
+##### `craftguide.progressive_show_usages(item, player)`
```

Comment:
:-1: Showing usages sounds like cheating the progressive mode to me. Please remove it.

Reply:
Note that this function does nothing by default (usages never shown). In some cases you will want to show usages for an item (like if the item cannot be used to craft secretstuff). Now I realize there's a better way to do this.

Reply:
We don't actually need that function. Usages are recipes, when you look at usages you're just looking at a subset of the available recipes (the ones with the clicked item as an input). So the recipe filter function should be used.

Test: enable progressive and put `default:wood` and `default:stick` in your inventory. You cannot see usages for any item except the stick. When you look at the usages for the stick you see a subset of the already shown recipes (the ones with a stick as an input). You can't see any items that require inputs that are not in your inventory.